packages/astro/src/server/clerk-middleware.ts

```diff
@@ -218,14 +224,38 @@ Check if signInUrl is missing from your configuration or if it is not an absolut
    PUBLIC_CLERK_SIGN_IN_URL='SOME_URL'
    PUBLIC_CLERK_IS_SATELLITE='true'`;
 
-function decorateAstroLocal(req: Request, context: APIContext, requestState: RequestState) {
+function decorateAstroLocal(clerkRequest: ClerkRequest, context: APIContext, requestState: RequestState) {
   const { reason, message, status, token } = requestState;
   context.locals.authToken = token;
   context.locals.authStatus = status;
   context.locals.authMessage = message;
   context.locals.authReason = reason;
-  context.locals.auth = () => getAuth(req, context.locals);
-  context.locals.currentUser = createCurrentUser(req, context);
+  context.locals.auth = () => {
+    const authObject = getAuth(clerkRequest, context.locals);
+
+    const clerkUrl = clerkRequest.clerkUrl;
+
+    const redirectToSignIn: RedirectFun<Response> = (opts = {}) => {
+      const devBrowserToken =
+        clerkRequest.clerkUrl.searchParams.get(constants.QueryParameters.DevBrowser) ||
+        clerkRequest.cookies.get(constants.Cookies.DevBrowser);
+
+      return createRedirect({
+        redirectAdapter,
+        devBrowserToken: devBrowserToken,
+        baseUrl: clerkUrl.toString(),
+        publishableKey: getSafeEnv(context).pk!,
+        signInUrl: requestState.signInUrl,
+        signUpUrl: requestState.signUpUrl,
+      }).redirectToSignIn({
+        returnBackUrl: opts.returnBackUrl === null ? '' : opts.returnBackUrl || clerkUrl.toString(),
```

Comment:
Does the `redirectToSignUp` function commonly used?

Reply:
We've build it, but no public API consumes it. and this implementation aims to mirror the API from @clerk/nextjs.

Reply:
Thanks for answering!